app/src/main/java/es/wolfi/app/passman/SJCLCrypto.java

```diff
@@ -21,10 +21,159 @@
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
 public class SJCLCrypto {
-    public static native String decryptString(String cryptogram, String key) throws Exception;
+    //public static native String decryptString(String cryptogram, String key) throws Exception;
+
+    private static final String ALGORITHM = "AES/CCM/NoPadding";
+    private static final int TAG_LENGTH_BIT = 64;
+    private static final int SALT_LENGTH_BYTE = 8;
+    private static final int ITER = 1000;
+    private static final int KEY_LENGTH_BIT = 256;
 
-    static {
+    /*static {
         System.loadLibrary("passman-lib");
+    }*/
+
+    public static JSONObject encryptString(String toencrypt, String stringkey)
+    {
+        try {
+
+            SecureRandom random = new SecureRandom();
+
+            byte[] ivBytes;
+            byte bytes[] = new byte[SALT_LENGTH_BYTE];
+            random.nextBytes(bytes);
+            byte[] saltBytes = bytes;
+
+            PBEKeySpec keySpec = new PBEKeySpec(
+                    stringkey.toCharArray(),
+                    saltBytes,
+                    ITER,
+                    KEY_LENGTH_BIT);
+
+            SecretKeyFactory secretKeyFactory = SecretKeyFactory.getInstance("PBKDF2withHmacSHA256");
+            SecretKey secretKey = secretKeyFactory.generateSecret(keySpec);
+
+            SecretKeySpec secretSpec = new SecretKeySpec(secretKey.getEncoded(),"AES");
+
+            Cipher aes = Cipher.getInstance(ALGORITHM);
+
+            aes.init(Cipher.ENCRYPT_MODE,secretSpec);
+
+            AlgorithmParameters algorithmParameters = aes.getParameters();
+            ivBytes = algorithmParameters.getParameterSpec(IvParameterSpec.class).getIV();
+
+            byte[] encryptedTextBytes = aes.doFinal(toencrypt.getBytes("UTF-8"));
+
+            String saltString = Base64.encodeToString(saltBytes,Base64.NO_WRAP);
+            String ivString = Base64.encodeToString(ivBytes,Base64.NO_WRAP );
+            String encryptedString = Base64.encodeToString(encryptedTextBytes,Base64.NO_WRAP);
+
+            JSONObject jsonResult = new JSONObject();
+
+            jsonResult.put("iv", ivString);
+            jsonResult.put("v", 1);
+            jsonResult.put("iter", ITER);
+            jsonResult.put("ks", KEY_LENGTH_BIT);
+            jsonResult.put("ts", TAG_LENGTH_BIT);
+            jsonResult.put("mode", "ccm");
+            jsonResult.put("adata", "");
+            jsonResult.put("cipher", "aes");
+            jsonResult.put("salt", saltString);
+            jsonResult.put("ct", encryptedString);
+
+            return jsonResult;
+
+        }
+        catch (Exception ex)
+        {
+            Log.d("SJCL", "Could not encrypt: " + ex.toString());
+        }
+        return null;
+    }
+
+    public static String decryptString(String todecrypt, String stringkey)
+    {
+        try {
```

Comment:
This approach will drop support for any android device before android 8 (api 26).
Also if at the end we choose this approach instead of the native one (which i found a different library that seems easier to implement than openssl called wolfssl) this should throw an exception instead of null, i rather catch an exception that compare data types to check if decryption was successfully or failed.

I already explain some caveats of this approach instead of the exception one in another comment where this function is called.